### ISSUE TITLE
Using Gickup as an all-org downloader and updater?

### ISSUE THREAD

**Issue by colindean** [cooperspencer/gickup#52]
A tool I've written and rewritten at seemingly every company I've been at is a tool that checks out all of the repos for a project. Oftentimes, this means "checkout all of the repos in an org".

I think Gickup could be the replacement for this, which I've usually written as a Ruby script but one time published [this bash+jq script](https://gist.github.com/colindean/ffd30db8d868e71438db) that handled the one-time clone operation.

It's pretty clear that a properly-configured Gickup could download all repos in a single org: I've not tested it, but simply set the `username` option to the org!

Could Gickup also run in the background, on its cron schedule, and _fetch_ updates so that a dev's copy of all of the org's source has the latest commits in the index so they always have the most up to date changes, even if their working directory is dirty? Put more simply, **when Gickup updates, does it expect a clean working directory?**

This could be a really cool use case for Gickup: keep your local copies up to date and you need only merge or rebase, e.g. `git merge origin/master` and you've already got the latest commits as of the last time Gickup ran.

**Comment by cooperspencer** [comment]
I just tested your case.
I used the org name as the user and it cloned  all the repos from there. Sadly gickup expects a clean working directory, otherwise it deletes the repo and clones it again. I can implement that it wouldn't remove the repo if the pull fails and just fetches the changes from the origin. Would that help you out?

**Comment by colindean** [comment]
Yeah, I think a fetch would be appropriate, with a pull if the working copy is clean.

**Comment by cooperspencer** [comment]
just pushed a new branch [fetch_working_dir](https://github.com/cooperspencer/gickup/tree/fetch_working_dir). You can test if that suits your needs.

**Comment by colindean** [comment]
Will do, I'll probably check this out at work this week.

**Comment by cooperspencer** [comment]
does it work for you? I would push the changes into `main` if it works out for you

**Comment by colindean** [comment]
Sorry, I ended up taking off this week and am trying not to touch my work laptop. I'll try next week.

**Comment by cooperspencer** [comment]
No worries.
